app/controllers/friendships_controller.rb

```diff
@@ -1,6 +1,7 @@
 class FriendshipsController < ApplicationController
   def create
-    friend = Friendship.new(user_id: current_user.id, friend_id: User.find_by(uid: params[:uid]))
+    friend = User.find_by(uid: params[:uid])
+    Friendship.create(user_id: current_user.id, friend_id: friend.id)
```

Comment:
This won't work. `friend` is not saving the friendship but just the user we've found. Correct fix is this:
```suggestion
    friend = Friendship.new(user_id: current_user.id, friend_id: User.find_by(uid: params[:uid]).id)
```